lib/datadog/statsd/connection_cfg.rb

```diff
@@ -10,28 +10,31 @@ class ConnectionCfg
       DEFAULT_PORT = 8125
 
       def initialize(host: nil, port: nil, socket_path: nil)
+        initialize_with_constructor_args(host, port, socket_path) || initialize_with_env_vars || initialize_with_defaults
+      end
 
-        # Try with constructor args
-        if setup_with(host, port, socket_path)
-          return
-        end
+      def initialize_with_constructor_args(host, port, socket_path)
+        try_initialize_with(host, port, socket_path,
+          "Both host/port and socket_path constructor arguments are set.  Set only one or the other.",
+          )
+      end
 
-        # Try with env vars
-        if setup_with(
-            ENV['DD_AGENT_HOST'],
-            ENV['DD_DOGSTATSD_PORT'].nil? ? nil : ENV['DD_DOGSTATSD_PORT'].to_i,
-            ENV['DD_DOGSTATSD_SOCKET'])
-          return
-        end
+      def initialize_with_env_vars()
+        try_initialize_with(
+          ENV['DD_AGENT_HOST'],
+          ENV['DD_DOGSTATSD_PORT'].nil? ? nil : ENV['DD_DOGSTATSD_PORT'].to_i,
+          ENV['DD_DOGSTATSD_SOCKET'],
+          "Both $DD_AGENT_HOST/$DD_DOGSTATSD_PORT and $DD_DOGSTATSD_SOCKET are set.  Set only one or the other.",
```

Comment:
Minor: Note that `$DD_AGENT_HOST` actually means something in Ruby -- variables prepended with `$` are global (I think it's a carry over from perl...?). So it may be helpful to clarify that we mean the environment variables.